src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SinkIslandCheck.java

```diff
@@ -206,8 +203,29 @@ private boolean endNodeHasAmenityTypeToExclude(final AtlasObject object)
         final Edge edge = (Edge) object;
         final Node endNode = edge.end();
 
-        return Validators.isOfType(endNode, AmenityTag.class, AmenityTag.PARKING,
-                AmenityTag.PARKING_SPACE, AmenityTag.MOTORCYCLE_PARKING,
-                AmenityTag.PARKING_ENTRANCE);
+        return Validators.isOfType(endNode, AmenityTag.class, amenityValuesToExclude);
+    }
+
+    /**
+     * This function checks an edge to determine whether it has certain characteristics that signify
+     * to us that we do not want to keep examining this component of the network.
+     *
+     * @param edge
+     *            An Edge we're examining
+     * @return {@code true} if the edge is already flagged, has an amenity type we want to exclude,
+     *         or ends in a boundary node {@code false} otherwise
+     */
+    private boolean shouldQuitNow(final Edge edge)
+    {
+        // If the edge has already been flagged by another process then we can break out of the
+        // loop and assume that whether the check was a flag or not was handled by the other
+        // process
+        return this.isFlagged(edge.getIdentifier())
```

Comment:
Could this cause branching edges to be missed because another branch has already been flagged? Example:
![slide1](https://user-images.githubusercontent.com/24948563/46758236-37929180-cc81-11e8-80cf-7094740903ff.png)

Reply:
I think you're exactly right. I decided not to fix this in this round of enhancements for a couple of reasons:
- keeping the diff between pre-enhancement and post-enhancement runs manageable
- performance, since performing the entire search every time would be significantly more expensive
- in the case you mentioned above, and other cases I could think of, solving the flagged problem also solves the unflagged problem. That is, if the top edge in your diagram gets fixed to be a non-sink, then the unflagged edge will also be a non-sink. 

Definitely open to modifying this, though!

Reply:
To be clear, this logic was present in this check before this PR. I just moved it into a function because there were some other checks I wanted to perform at the same time.

Reply:
It is an excellent point that all the sink edges will be resolved if the flagged edges are fixed. So, this really shouldn't be an issue.